Sources/MapboxNavigation/NavigationMapView+VanishingRouteLine.swift

```diff
@@ -85,7 +85,7 @@ extension NavigationMapView {
         if currentStepProgress.distanceTraveled <= 0 {
             allRemainingPoints += currentLegSteps[currentLegProgress.stepIndex].dropLast().count
         } else if let startIndex = lineString.indexedCoordinateFromStart(distance: currentStepProgress.distanceTraveled)?.index,
-                  startIndex < lineString.coordinates.endIndex - 1 {
```

Comment:
Have you managed to reproduce the actual crash? Would it be possible to somehow simulate it in unit tests to make sure that we do not crash here?

Reply:
I can reproduce this issue in unit test. This crash could only happen when the `routePoints` is empty from an `empty` route with no coordinates inside each `RouteStep`. With a non-empty route, even the user hasn't started the routeProgress, the `routeRemainingDistancesIndex` won't be negative. I added the unit tests in the new commit.